compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3443,6 +3443,12 @@ public void visit(BLangXMLElementLiteral bLangXMLElementLiteral) {
 
         resultType = checkXmlSubTypeLiteralCompatibility(bLangXMLElementLiteral.pos, symTable.xmlElementType,
                                                          this.expType);
+
+        if (!Symbols.isFlagOn(resultType.flags, Flags.READONLY)) {
```

Comment:
Shouldn't this code be
```Java
if (Symbols.isFlagOn(resultType.flags, Flags.READONLY)) {
   markChildrenAsImmutable(bLangXMLElementLiteral);
}
```

Reply:
Would it make a difference? And I think the common case would be the current check evaluating to true (i.e., mutable result type).

Reply:
My opinion is that if lines are separated here, we lose the actual logic we need. Logic is not obvious anymore. Because these two lines are separated someone may put some lines in between in future and decide to return. At that time for some elements, we won't be calling `markChildrenAsImmutable`.

Reply:
I've updated the logic with 8e8807b since this is a single invocation. But regarding adding lines in between, IMO the logic here is quite straightforward and it is reasonable to expect whoever doing the change to understand the area they are changing when doing the changes.  :) Especially given that we've adopted a similar approach with a significant number of functions.